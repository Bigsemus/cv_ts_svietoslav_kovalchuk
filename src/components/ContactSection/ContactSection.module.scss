@import "../../utils/constants/styleVariable";

.interactiveElementHTML {
  cursor: url(../../assets/icon/hand-cursor-transparent-dark.svg), auto;
  &.interactiveElementHTMLDark {
    cursor: url(../../assets/icon/hand-cursor-transparent.svg), auto;
  }
}


.contactlink {
  font-weight: bold;
  text-decoration: none;
  color: $color-text;

  &.active {
    color: rgb(251, 251, 251);
  }

  &:hover,
  &:focus {
    color: $color-accordeon-hover;
  }
}