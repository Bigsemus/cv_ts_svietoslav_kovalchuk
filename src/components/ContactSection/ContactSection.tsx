import React, {useContext, useEffect, useMemo, useState} from 'react';
import address from "../../assets/icon/a-icon-address/Combined Shape.png";
import phone from "../../assets/icon/a-icon-phone/Combined Shape.png";
import email from "../../assets/icon/a-icon-email/Combined Shape.png";
import useTranslation from "../../Hooks/useTranslation";
import {ThemeContext} from "../../Context/contextTheme";
import classNames from "classnames/bind";
import classes from "./ContactSection.module.scss";
import useCursorPointerWithTheme from "../../Hooks/useCursorPointerWithTheme";
import FormikContact from "../FormikContact";

const ContactSection = () => {
    const {theme} = useContext(ThemeContext);
    const styleCursorPointer = useCursorPointerWithTheme();
    const cx = classNames.bind(classes);


    const defaultHoverValues = useMemo(() => {
        return {
            isLinkHover: false,
        };
    }, []);


    const [valueStylesWithTheme, setValueStylesWithTheme] = useState(defaultHoverValues);
    const link = cx('contactlink', {
        active: valueStylesWithTheme.isLinkHover,
    });

    useEffect(() => {
        setValueStylesWithTheme(
            theme.darkTheme === 'true'
                ? {...defaultHoverValues, isLinkHover: true}
                : {...defaultHoverValues, isLinkHover: false}
        )
    }, [theme, defaultHoverValues])

    return (
        <section id="contact" className="main__contact main-contact">
            <div className="wrapper">
                <h2 className="main__title">
                    {useTranslation('home-section-contact.title')}
                </h2>
                <hr className="main__line" />
                <div className="grid-markup">
                    <div className="grid-markup__main-grid-block grid-markup__main-grid-block--left">
                        <h3 className="main__title-little">
                            {useTranslation('home-section-contact.main__title-little')}
                        </h3>

                        <FormikContact />

                    </div>
                    <div className="grid-markup__main-grid-block grid-markup__main-grid-block--right">
                        <h3 className="main__title-little">
                            {useTranslation('home-section-contact.title-little2')}
                        </h3>
                        <p className="main-contact__sub-title-information">
                            {useTranslation('home-section-contact.sub-title')}
                        </p>
                        <div className="main-contact__data-company data-company">
                            <img src={address} alt="address-icon" />
                            <span className="data-company__item">
                                {useTranslation('home-section-contact.data-company__item1')}
                            </span>
                            <span className="data-company__item">
                                {useTranslation('home-section-contact.data-company__item2')}
                            </span>
                            <span className="data-company__item">
                                {useTranslation('home-section-contact.data-company__item3')}
                            </span>
                        </div>
                        <p className="main-contact__contact contact">
                            <img className="contact__logo" src={phone}
                                 alt="phone-icon" />
                            {useTranslation('home-section-contact.phone')}
                            <a
                                className={classNames(link, styleCursorPointer)}
                                href="tel:+380500890028"
                            >
                                &nbsp;+380500890028
                            </a>
                        </p>
                        <p className="main-contact__contact">
                            <img className="contact__logo" src={email}
                                 alt="phone-icon" />
                            {useTranslation('home-section-contact.email')}
                            <a
                                className={classNames("contact__link main__link-text main__link-text--primary", styleCursorPointer)}
                                href="mailto:Svietoslav.Kovalchuk@gmail.com">&nbsp;Svietoslav.Kovalchuk@gmail.com</a>
                        </p>
                    </div>
                </div>
            </div>
        </section>
    );
};

export default ContactSection;