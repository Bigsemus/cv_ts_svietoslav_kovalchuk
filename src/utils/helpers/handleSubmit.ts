import React from "react";

 const handleSubmit: React.ReactEventHandler = (event) => {
    //event.preventDefault()
    // const matchedInputsValue = {};
    // const initialValues = {
    //     'nameInput': '',
    //     'emailInput': '',
    //     'lastNameInput': '',
    //     'phoneInput': '',
    //     'radio': '',
    //     'messageTextArea': '',
    //     'conditionsCheckbox': '',
    //     refresh: (name: string, value: string) => {
    //         Object.assign(matchedInputsValue, {[name]: value});
    //     }
    // }
    //
    // const inputs = document
    //     .querySelectorAll('input, textarea') as unknown as HTMLElement[] | any;
    // for (let q = 1; q < inputs.length; q++) {
    //     initialValues.refresh(inputs[q]?.name, inputs[q]?.value)
    // }
    //  console.log(matchedInputsValue)
    //return matchedInputsValue
}

export default handleSubmit