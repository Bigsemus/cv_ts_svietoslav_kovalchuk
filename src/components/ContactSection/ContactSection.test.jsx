import { fireEvent, render, screen } from "@testing-library/react";
import ContactSection from "./ContactSection";
import handleSubmit from "../../utils/helpers/handleSubmit";

test('render ContactSection', () => {
  const { container } = render(<ContactSection />);
  expect(container).toMatchSnapshot();
})

test('fireEvent', () => {
  render(<ContactSection />);
  //jest.spyOn(ContactSection, handleSubmit)
  const matchedInputsValue = {"conditionsCheckbox": "on", "emailInput": "", "lastNameInput": "", "messageTextArea": "", "phoneInput": "", "radio": "3"};
  // const spy = jest.spyOn(handleSubmit);
  const button = screen.getByRole('button', { name: /submit/i })
  const clickEvent = new MouseEvent('click');
  Object.assign(clickEvent, { preventDefault: jest.fn() });
  fireEvent(button, clickEvent);

  expect(clickEvent.preventDefault).toHaveBeenCalledTimes(0);

})