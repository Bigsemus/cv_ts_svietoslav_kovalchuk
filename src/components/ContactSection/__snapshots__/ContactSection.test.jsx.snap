// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ContactSection 1`] = `
<div>
  <section
    class="main__contact main-contact"
    id="contact"
  >
    <div
      class="wrapper"
    >
      <h2
        class="main__title"
      >
        Contact
      </h2>
      <hr
        class="main__line"
      />
      <div
        class="grid-markup"
      >
        <div
          class="grid-markup__main-grid-block grid-markup__main-grid-block--left"
        >
          <h3
            class="main__title-little"
          />
          <form
            class="main-contact__contact-form contact-form"
            name="ContactForm"
          >
            <div
              class="contact-form__row contact-form__row--first"
            >
              <label>
                First name
                 *
                <input
                  class="contact-form__input"
                  name="nameInput"
                  placeholder="  John"
                  type="text"
                />
              </label>
              <label>
                Email
                 *
                <input
                  class="contact-form__input"
                  name="emailInput"
                  placeholder="  example@email.com"
                  type="email"
                />
              </label>
            </div>
            <div
              class="contact-form__row"
            >
              <label>
                Last name
                <input
                  class="contact-form__input"
                  name="lastNameInput"
                  placeholder="  Johnson"
                  type="text"
                />
              </label>
              <label>
                Phone
                 *
                <input
                  class="contact-form__input"
                  name="phoneInput"
                  placeholder="  + 000 123 456 789"
                  type="tel"
                />
              </label>
            </div>
            <div
              class="contact-form__sex-radio sex-radio"
            >
              <p
                class="sex-radio__title"
              >
                Choose option:
              </p>
              <div
                class="sex-radio__item"
              >
                <input
                  id="radio-1"
                  name="radio"
                  type="radio"
                  value="1"
                />
                <label
                  for="radio-1"
                >
                  Male
                </label>
              </div>
              <div
                class="sex-radio__item"
              >
                <input
                  id="radio-2"
                  name="radio"
                  type="radio"
                  value="2"
                />
                <label
                  for="radio-2"
                >
                  Female
                </label>
              </div>
              <div
                class="sex-radio__item"
              >
                <input
                  id="radio-3"
                  name="radio"
                  type="radio"
                  value="3"
                />
                <label
                  for="radio-3"
                >
                  Other
                </label>
              </div>
            </div>
            <textarea
              class="contact-form__textarea"
              name="messageTextArea"
              placeholder="Message"
            />
            <div
              class="contact-form__checkbox"
            >
              <input
                id="checkbox-1"
                name="conditionsCheckbox"
                type="checkbox"
              />
              <label
                for="checkbox-1"
              >
                Terms & Conditions
              </label>
            </div>
            <button
              class="contact-form__submit"
              name="submit"
              role="button"
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
        <div
          class="grid-markup__main-grid-block grid-markup__main-grid-block--right"
        >
          <h3
            class="main__title-little"
          >
            Contact information
          </h3>
          <p
            class="main-contact__sub-title-information"
          >
            Feel free to contact us for any of your questions, we’ll get back to you as soon as possible
          </p>
          <div
            class="main-contact__data-company data-company"
          >
            <img
              alt="address-icon"
              src="Combined Shape.png"
            />
            <span
              class="data-company__item"
            >
              Epam
            </span>
            <span
              class="data-company__item"
            >
              Barykadna, 16
            </span>
            <span
              class="data-company__item"
            >
              Ukraine, Dnipro
            </span>
          </div>
          <p
            class="main-contact__contact contact"
          >
            <img
              alt="phone-icon"
              class="contact__logo"
              src="Combined Shape.png"
            />
            Phone:
            <a
              class="contact__link main__link-text main__link-text--dark"
              href="tel:+380500890028"
            >
               +380500890028
            </a>
          </p>
          <p
            class="main-contact__contact"
          >
            <img
              alt="phone-icon"
              class="contact__logo"
              src="Combined Shape.png"
            />
            Email:
            <a
              class="contact__link main__link-text main__link-text--primary"
              href="mailto:Svietoslav.Kovalchuk@gmail.com"
            >
               Svietoslav.Kovalchuk@gmail.com
            </a>
          </p>
        </div>
      </div>
    </div>
  </section>
</div>
`;
